File: apps/sandbox-go/service/execService.go
package service

import (
	"sandbox-go/executor"
	"sandbox-go/utils"
)

type ExecService struct {
	executor *executor.ProcessExecutor
}

func NewExecService(executor *executor.ProcessExecutor) *ExecService{
	return &ExecService{executor: executor}
}

func (s *ExecService)Execute(req utils.ExecRequest) utils.ExecResponse{
	if req.Command == ""{
		return utils.ExecResponse{
			Stdout: "",
			Stderr: "",
			ExitCode: -1,
			Error: "Command is required",
		}
	}

	return s.executor.Exec(req)
}

File: apps/sandbox-ts/service/execService.ts
import { ProcessExecutor } from "../executor/processExecutor";
import { ExecRequest, ExecResponse } from "../utils/types";

export class ExecService {
    private executor: ProcessExecutor;

    constructor(executor:ProcessExecutor){
        this.executor = executor
    }

    async execute(req:ExecRequest): Promise<ExecResponse>{
        if(!req.command){
            return {
                stdout: "",
                stderr: "",
                exitCode: -1,
                error: "Command is required"
            }
        }
        return this.executor.exec(req)
    }
}